### PR TITLE
Plugins: Avoid returning 404 for `AutoEnabled` apps

### LINKED ISSUE #93243: Setting `autoEnabled` in an app plugin's `plugin.json` does not enable backend routes for subsequent orgs
### What happened?

I have an app plugin with `"backend": true` and `"autoEnabled": true`. It works as intended on the primary org. Nothing appears in the logs to suggest anything is broken.

Newly-created orgs get the app plugin's frontend code, and the plugin is visible and enabled on the plugins page, but the API routes report 404 and "Plugin not found". This appears in the log (edited for brevity): `level=info msg="Request Completed" method=GET path=/api/plugins/<my_plugin_id>/resources/reports status=404 handler=/api/plugins/:pluginId/resources/* status_source=server`

### What did you expect to happen?

If new orgs automatically load the app's frontend code, the Go routes should also load.

### Did this work before?

Tested in 10.3.3 and 11.2.0, neither worked. I have not tried farther back.

### How do we reproduce it?

1. Fetch a copy of the [app-with-backend example](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/app-with-backend)
2. Add `"autoEnabled": true,` to `plugin.json` and save
3. Build the client: `npm install && npm build`, then the server: `go get -u github.com/grafana/grafana-plugin-sdk-go && go mod tidy && mage -v`, then run the app plugin within Docker: `npm run server`
4. Login using the default admin credentials. The app plugin's `/ping` route works as expected.
5. Create a new org and switch context to it. You will still see the app plugin's frontend, but when you navigate to it the `/ping` route will fail.

Bonus:
1. Using your new org, go to the plugins page
2. The app plugin is listed as enabled, but it's not really fully enabled
3. Disable the plugin, then enable it again
4. The API routes of the `autoEnabled` app plugin are now actually enabled and should work as expected.

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 10.3.3+
OS: within Docker on MacOS
Browser: Chrome, recent

### Grafana platform?

Docker

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by wbrowne**
**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/93243

### CODE REVIEW

File: pkg/api/plugin_checks.go [hunk 1]
```diff
@@ -28,7 +28,13 @@ func checkAppEnabled(pluginStore pluginstore.Store, pluginSettings pluginsetting
...
 		if err != nil {
 			if errors.Is(err, pluginsettings.ErrPluginSettingNotFound) {
-				c.JsonApiErr(http.StatusNotFound, "Plugin not found", nil)
+				// If the plugin is auto enabled, we don't want to return an error because auto enabling allows us
+				// to enable plugins that are not explicitly configured.
+				if p.AutoEnabled {
+					return
+				}
+
+				c.JsonApiErr(http.StatusNotFound, "Plugin setting not found", nil)
```
Comment (wbrowne) [inline pkg/api/plugin_checks.go:37]: Note: I also updated the wording here as it was confusing initially when doing an RCA. Not sure if it was an oversight originally or on purpose

Comment (andresmgot) [reply pkg/api/plugin_checks.go:37]: Looks like a copy-paste error from line 18 so this is fine

**Review by andresmgot — COMMENTED** [review]
There is a very similar function in https://github.com/grafana/grafana/blob/pluginSDK249/pkg/services/navtree/navtreeimpl/applinks.go#L28 for getting if an app is enabled or not. We should probably refactor it so both functions use the same logic. Can you give that a try? If it doesn't work nicely we can leave it as it is since both functions now has the same logic

**Comment by wbrowne** [comment]
> There is a very similar function in https://github.com/grafana/grafana/blob/pluginSDK249/pkg/services/navtree/navtreeimpl/applinks.go#L28 for getting if an app is enabled or not. We should probably refactor it so both functions use the same logic. Can you give that a try? If it doesn't work nicely we can leave it as it is since both functions now has the same logic

Good idea! I took a look and unfortunately since `GetPluginSettings` and `GetPluginSettingByPluginID` return two different type of structs, it wouldn't be as easy as expected 😞

**Review by andresmgot — APPROVED** [review]
LGTM

**Pull request merged**